models/contacts.js

```diff
@@ -1,19 +1,115 @@
-// const fs = require('fs/promises')
+const fs = require("fs");
+const fsPromise = require("fs/promises");
+const path = require("path");
 
-const listContacts = async () => {}
+// * Declare file path
+const contactsPath = path.normalize("models/contacts.json");
 
-const getContactById = async (contactId) => {}
+const getContacts = async () => {
+  try {
+    const data = await fsPromise.readFile(contactsPath, { encoding: "utf8" });
+    return data;
+  } catch (err) {
+    return err;
+  }
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async (contactId) => {
+  const contacts = await getContacts();
+  const readedData = JSON.parse(contacts);
+  const findedContact = readedData.find((obj) => obj.id === contactId);
+  if (findedContact) {
+    return findedContact;
+  } else {
+    return null;
+  }
+};
 
-const addContact = async (body) => {}
+const removeContact = async (contactId) => {
+  const contacts = await getContacts();
+  const readedData = JSON.parse(contacts);
+  const findedContactIndex = readedData.findIndex(
+    (obj) => obj.id === contactId
+  );
+  if (findedContactIndex !== -1) {
+    readedData.splice(findedContactIndex, 1);
+    fs.writeFile(contactsPath, JSON.stringify(readedData), (err) => {
+      if (err) return err;
+    });
+    return true;
+  } else {
+    return false;
+  }
+};
 
-const updateContact = async (contactId, body) => {}
+const addContact = async ({ name, email, phone }) => {
+  const contacts = await getContacts();
+  const readedData = JSON.parse(contacts);
+  const newContact = {
+    id: `${readedData.length + 1}`,
```

Comment:
use `uuid` to generate random id. Or any other library. You can use Date object for that if you like but your way of doing it is wrong. You can easily add two contacts with same index which will lead to a lot of problems :)